kr8s/_portforward.py

```diff
@@ -130,21 +143,30 @@ async def run_forever(self) -> None:
         """
         async with self:
             with contextlib.suppress(asyncio.CancelledError):
-                await self.server.serve_forever()
+                for server in self.servers:
+                    await server.serve_forever()
 
     @asynccontextmanager
     async def _run(self) -> int:
-        """Start the port forward and yield the local port."""
-        self.server = await asyncio.start_server(
-            self._sync_sockets, port=self.local_port, host="0.0.0.0"
-        )
-        async with self.server:
-            await self.server.start_serving()
-            for sock in self.server.sockets:
-                if sock.family == socket.AF_INET:
-                    yield sock.getsockname()[1]
-            self.server.close()
-            await self.server.wait_closed()
+        """Start the port forward for multiple bind addresses and yield the local port."""
+        self.servers = []
+        for address in self.address:
+            server = await asyncio.start_server(
+                self._sync_sockets, port=self.local_port, host=address
+            )
+            self.servers.append(server)
+
+        try:
+            for server in self.servers:
+                async with server:
+                    await server.start_serving()
+                    for sock in server.sockets:
+                        if sock.family == socket.AF_INET:
+                            yield sock.getsockname()[1]
```

Comment:
I'm not sure this works as expected any more because with multiple servers we will yield multiple times, which doesn't make sense for a context manager.

Once we have called `start_serving` on all the servers we probably just need to yield the port number for the first server.